csrc/device_lower/pass/unroll.cpp

```diff
@@ -167,10 +167,17 @@ void UnrollPass::dispatch(Expr* expr) {
       }
     }
 
+    // CpAsync uses inline predicate but still needs a IfThenElse predicate
+    // to avoid illegal memory access, see test
+    // FusionCpAsyncPredicateAvoidIllegalMemoryAccess
     if (lower_utils::supportInlinePredicate(expr)) {
```

Comment:
Can we make `lower_utils::supportInlinePredicate(expr)` return false if `needs_predicate_smem_access` is true?

Reply:
What about changing to the following?  The logic is: add inline predicate if the syntax supports it, then check wheter we should also add IfThenElse predicate.

```
    // Try to use inline predicate if possible.
    // If don't need shared memory predicate, just use inline predicate.
    // otherwise, also need to add IfThenElse predicate to avoid illegal memory
    // access. see test FusionCpAsyncPredicateAvoidIllegalMemoryAccess
    if (lower_utils::supportInlinePredicate(expr)) {
      expr_with_predicate = expr_with_predicate->withPredicate(pred);
      if (!GpuLower::current()
               ->predicateElimination()
               .needsSharedMemoryPredicate(expr)) {
        registerReplace(expr, expr_with_predicate);
        return;
      }
    }
```

Reply:
Looks good to me.